src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const length = this.length;
+    const separatorValue = separator;
+    let result = '';
+
+    for (let i = 0; i < length; i++) {
+      let element = this[i];
+
+      if (element === undefined || element === null) {
+        element = '';
+      }
+
+      if (i === length - 1) {
+        result += element;
+        break;
```

Comment:
no need for a break here. Basically the loop will break itself, as it is the last iteration